api/src/test/java/com/epam/pipeline/manager/pipeline/RunScheduleManagerTest.java

```diff
@@ -16,301 +16,282 @@
+                .when(mockRunScheduleDao).loadAllRunSchedulesBySchedulableIdAndType(RUN_ID, ScheduleType.PIPELINE_RUN);
 
-        runScheduleManager.deleteSchedulesForRunByPipeline(testPipeline.getId());
+        doReturn(Arrays.asList(createPipelineRun(RUN_ID, testPipeline.getId())))
+                .when(mockPipelineRunManager).loadAllRunsByPipeline(testPipeline.getId());
 
-        final List<RunSchedule> loadRunSchedule = runScheduleManager
-                .loadAllSchedulesBySchedulableId(RUN_ID, ScheduleType.PIPELINE_RUN);
-        assertEquals(0, loadRunSchedule.size());
-        Mockito.verify(runScheduler, Mockito.times(2)).unscheduleRunSchedule(any());
+        runScheduleManager.deleteSchedulesForRunByPipeline(testPipeline.getId());
 
+        verify(mockRunScheduler, times(2)).unscheduleRunSchedule(any());
+        verify(mockRunScheduleDao).deleteRunSchedulesForRunByPipeline(anyLong());
     }
 
     @Test(expected = IllegalArgumentException.class)
-    @WithMockUser(username = USER_OWNER)
     public void loadRunScheduleByWrongId() {
-        final RunSchedule runSchedule =
-            runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
-                    Collections.singletonList(testRunScheduleVO)).get(0);
-        runScheduleManager.loadSchedule(runSchedule.getSchedulableId());
-    }
+        doReturn(Optional.empty()).when(mockRunScheduleDao).loadRunSchedule(anyLong());
 
-    private PipelineRunScheduleVO getRunScheduleVO(final RunScheduledAction action, final String cronExpression) {
-        PipelineRunScheduleVO runScheduleVO = new PipelineRunScheduleVO();
-        runScheduleVO.setTimeZone(TIME_ZONE.getDisplayName());
-        runScheduleVO.setCronExpression(cronExpression);
-        runScheduleVO.setAction(action);
-        return runScheduleVO;
+        runScheduleManager.loadSchedule(1L);
     }
 
-    private void createPipelineRun(final Long runId, final Long pipelineId) {
+
+    private PipelineRun createPipelineRun(final Long runId, final Long pipelineId) {
         final AbstractCloudRegion cloudRegion = ObjectCreatorUtils.getDefaultAwsRegion();
-        regionDao.create(cloudRegion);
         PipelineRun run = ObjectCreatorUtils.createPipelineRun(runId, pipelineId, null, cloudRegion.getId());
         run.setStatus(TaskStatus.RUNNING);
         run.getInstance().setSpot(false);
-        pipelineRunDao.createPipelineRun(run);
+        return run;
     }
 
-    private RunConfiguration createRunConfiguration() {
-        //create
-        RunConfigurationEntry entry =
-                ObjectCreatorUtils.createConfigEntry("entry", true, null);
-
-        RunConfiguration configuration =
-                ObjectCreatorUtils.createConfiguration(TEST_NAME, null, null,
-                        USER_OWNER, Collections.singletonList(entry));
-        return configurationDao.create(configuration);
+    private RunSchedule createRunSchedule(final Long id, final Long schedulableId,
```

Comment:
It seems like there is only two repeating invocations of the method. Can we extract two class fields out of this and use them in all the tests?

![image](https://user-images.githubusercontent.com/16464586/109780061-accadc80-7c17-11eb-9b75-b95f1354bd92.png)

Something like

```java
private final RunSchedule runPauseSchedule = ...
private final RunSchedule runResumeSchedule = ...